service/src/main/java/io/camunda/service/search/filter/ProcessInstanceFilter.java

```diff
@@ -7,103 +7,136 @@
     boolean running,
     boolean active,
     boolean incidents,
     boolean finished,
     boolean completed,
     boolean canceled,
     boolean retriesLeft,
-    List<VariableValueFilter> variableFilters,
-    DateValueFilter startDateFilter,
-    DateValueFilter endDateFilter)
+    String errorMessage,
+    String activityId,
+    DateValueFilter startDate,
+    DateValueFilter endDate,
+    List<String> bpmnProcessIds,
+    List<Integer> processDefinitionVersions,
+    ProcessInstanceVariableFilter variable,
+    List<String> batchOperationIds,
+    List<Long> parentProcessInstanceKeys,
+    List<String> tenantIds)
     implements FilterBase {
 
   public static final class Builder implements ObjectBuilder<ProcessInstanceFilter> {
 
-    private List<Long> processInstanceKeys;
     private boolean running;
     private boolean active;
     private boolean incidents;
     private boolean finished;
     private boolean completed;
     private boolean canceled;
     private boolean retriesLeft;
-    private List<VariableValueFilter> variableFilters;
-    private DateValueFilter startDateFilter;
-    private DateValueFilter endDateFilter;
-
-    public Builder processInstanceKeys(final Long value, final Long... values) {
-      return processInstanceKeys(collectValues(value, values));
+    private String errorMessage;
+    private String activityId;
+    private DateValueFilter startDate;
+    private DateValueFilter endDate;
+    private List<String> bpmnProcessIds;
+    private List<Integer> processDefinitionVersions;
+    private ProcessInstanceVariableFilter variable;
+    private List<String> batchOperationIds;
+    private List<Long> parentProcessInstanceKeys;
+    private List<String> tenantIds;
+
+    public Builder running(final boolean running) {
+      this.running = running;
+      return this;
     }
 
-    public Builder processInstanceKeys(final List<Long> values) {
-      processInstanceKeys = addValuesToList(processInstanceKeys, values);
+    public Builder running() {
+      running = false;
       return this;
     }
 
-    public Builder running() {
-      running = true;
+    public Builder active(final boolean active) {
+      this.active = active;
       return this;
     }
 
     public Builder active() {
-      active = true;
+      active = false;
+      return this;
+    }
+
+    public Builder incidents(final boolean incidents) {
+      this.incidents = incidents;
       return this;
     }
 
     public Builder incidents() {
-      incidents = true;
+      incidents = false;
+      return this;
+    }
+
+    public Builder finished(final boolean finished) {
+      this.finished = finished;
       return this;
     }
 
     public Builder finished() {
-      finished = true;
+      finished = false;
+      return this;
+    }
+
+    public Builder completed(final boolean completed) {
+      this.completed = completed;
       return this;
     }
 
     public Builder completed() {
```

Comment:
I think this is confusing. The user doesn't know what happens in this case. Maybe it is better to have only one method like the 
```java
public Builder completed(final boolean completed) {
```
Same for all other boolean builder methods.